src/H5HFcache.c

```diff
@@ -742,7 +742,7 @@ H5HF__cache_hdr_serialize(const H5F_t *f, void *_image, size_t H5_ATTR_NDEBUG_UN
     HDassert(len == hdr->heap_size);
 
     /* Set the shared heap header's file context for this operation */
-    hdr->f = f;
+    hdr->f = (H5F_t *)f; /* TODO: should not cast away const! */
```

Comment:
Please don't cast away const, it's hiding the underlying design issue, which should be fixed.